models/placement.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const sequelize = require('../config/sequelize')
+const {reduce, replace} = require('lodash')
+
+class Placement {
+  static get QUERY () {
+    return 'SELECT ' +
+      'CAST(to_char(AVG(scores.unaware), \'FM9.999\') AS NUMERIC) as unaware, ' +
+      'CAST(to_char(AVG(scores.curious), \'FM9.999\') AS NUMERIC) as curious, ' +
+      'CAST(to_char(AVG(scores.follower), \'FM9.999\') AS NUMERIC) as follower, ' +
+      'CAST(to_char(AVG(scores.guide), \'FM9.999\') AS NUMERIC) as guide, ' +
+      'CAST(to_char(AVG(scores.unaware * (scores.confidence/100)), \'FM9.999\') AS NUMERIC) as weighted_unaware, ' +
+      'CAST(to_char(AVG(scores.curious * (scores.confidence/100)), \'FM9.999\') AS NUMERIC) as weighted_curious, ' +
+      'CAST(to_char(AVG(scores.follower * (scores.confidence/100)), \'FM9.999\') AS NUMERIC) as weighted_follower, ' +
+      'CAST(to_char(AVG(scores.guide * (scores.confidence/100)), \'FM9.999\') AS NUMERIC) as weighted_guide ' +
+      'FROM events ' +
+      'INNER JOIN scores as scores ON scores.uri = events.uri ' +
+      'WHERE events.user_id = :user_id AND events.created_at > current_timestamp - interval \'90\' day'
+  }
+
+  /**
+   * @param {User} user
+   */
+  constructor (user) {
+    this.user = user
+  }
+
+  /**
+   * @returns {Promise}
+   */
+  calculate () {
+    let self = this
+    return sequelize()
+      .query(Placement.QUERY, {replacements: {user_id: this.user.id}, type: sequelize().QueryTypes.SELECT})
+      .then(results => {
+        self._values = results[0]
+        return self
+      })
+  }
+
+  get values () {
+    return typeof this._values === 'undefined' ? {} : {
+      unaware: this._values['unaware'] || 0,
```

Comment:
0 is invalid. Will this be okay?

Reply:
At this point, unaware is the average of all the content unaware values. If it's missing for some reason (and I can't think of why), 0 should be ok. Placement compares each of them and takes the highest.

Reply:
And if the highest (somehow) ends up being 0, it will fail down the pipeline, right?